azurerm/internal/ar/client.go

```diff
@@ -10,26 +10,36 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/hashicorp/go-azure-helpers/sender"
 	"github.com/hashicorp/terraform/httpclient"
-	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 	"github.com/terraform-providers/terraform-provider-azurerm/version"
 )
 
-func ConfigureClient(client *autorest.Client, auth autorest.Authorizer, partnerId string, skipProviderReg bool) {
-	setUserAgent(client, partnerId)
-	client.Authorizer = auth
-	client.Sender = sender.BuildSender("AzureRM")
-	client.PollingDuration = 60 * time.Minute
-	client.SkipResourceProviderRegistration = skipProviderReg
-	client.RequestInspector = azure.WithCorrelationRequestID(azure.CorrelationRequestID())
+type ClientOptions struct {
+	Authorizer                 autorest.Authorizer
```

Comment:
since there's currently 3 authorizers (and about to be a 4th) we need to be more descriptive with this:

```suggestion
	ResourceManagerAuthorizer                 autorest.Authorizer
```